package/src/components/MessageList/MessageList.tsx

```diff
@@ -670,34 +656,32 @@ const MessageListWithContext = <
         threadList={threadList}
       />
     );
-    return wrapMessageInTheme ? (
-      <>
-        <ThemeProvider mergedStyle={modifiedTheme}>
-          <View
-            style={[shouldApplyAndroidWorkaround ? styles.invertAndroid : undefined]}
-            testID={`message-list-item-${index}`}
-          >
-            {shouldApplyAndroidWorkaround && renderDateSeperator}
+
+    return (
+      <View
+        style={[shouldApplyAndroidWorkaround ? styles.invertAndroid : undefined]}
+        testID={`message-list-item-${index}`}
+      >
+        {insertInlineUnreadIndicator ? <InlineUnreadIndicator /> : null}
+        {message.type === 'system' ? (
+          <MessageSystem
+            message={message}
+            style={[{ paddingHorizontal: screenPadding }, messageContainer]}
+          />
+        ) : wrapMessageInTheme ? (
+          <ThemeProvider mergedStyle={modifiedTheme}>
+            <View testID={`message-list-item-${index}`}>
+              {renderDateSeperator}
```

Comment:
I think that this and the same one below it would break if we pass `inverted={false}` on `Android`. Their `y-axis` orientation would be correct, however they would be in the wrong order.

Would you mind checking that ?

If it's indeed broken, something like:

```suggestion
              {shouldApplyAndroidWorkaround && renderDateSeperator}
              {renderMessage}
              {!shouldApplyAndroidWorkaround && renderDateSeperator}
```
should do the trick.